cudamat/cudamat.py

```diff
@@ -1501,6 +1503,25 @@ def where(condition_mat, if_mat, else_mat, target=None):
     return target
 
 
+def correlate(mat, kernel, target=None):
+    """
+    Apply a rectangular kernel matrix to the matrix mat.
```

Comment:
Better: "Cross-correlate a matrix with a kernel matrix."
"Apply" may be a bit unspecific :)